internal/dataplane/address_finder.go

```diff
@@ -72,6 +82,28 @@ func (a *AddressFinder) GetAddresses() ([]string, error) {
 	return nil, fmt.Errorf("data-plane addresses can't be retrieved: no valid method available")
 }
 
+// GetUDPAddresses provides a list of the UDP addresses which the data-plane is
+// listening on for ingress network traffic. Addresses can either be IP
+// addresses or hostnames. If UDP settings are not configured, falls back to GetAddresses().
+func (a *AddressFinder) GetUDPAddresses() ([]string, error) {
+	a.lock.RLock()
+	defer a.lock.RUnlock()
+
+	if len(a.overrideAddressesUDP) > 0 {
+		return a.overrideAddressesUDP, nil
+	}
+
+	if len(a.overrideAddresses) > 0 && a.addressGetter == nil {
+		return a.overrideAddresses, nil
+	}
+
+	if a.addressGetter != nil {
+		return a.addressGetter()
+	}
+
+	return a.GetAddresses()
```

Comment:
`a.GetAddresses` does the same thing as the code above, so it is redundant. Either call `a.GetAddresses` after checking `overrideAddressesUDP` (both uses RLock, so no deadlock risk) or removing this call and return error here,

Reply:
Removing this breaks the unit, I think without it we can't override. Can you clarify what you're suggesting instead?